### PR TITLE
Armagan - #TLI 02  - Single Demographic Selection

### PR DESCRIPTION
**Pull request by armaganngul**
Seperated the functionality of create_bias_data_points so we dont have to make jabilion single_check_column comparisons. 
ChatGPT suggested to put these files in a different file completely so will look into that too. I think also now we don't need evaluate_fairness() anymore in model.py? Cause @akshatoza1 made a seperate Fairlearn class?

### CODE REVIEW

File: backend/ml_model/use_cases/model.py [hunk 1]
```diff
@@ -16,55 +16,93 @@
...
-from ml_model.repository.fairness import FairnessEvaluator
 from ml_model.repository.data_preprocessing import DataProcessor
 
 current_dir = os.path.dirname(os.path.abspath(__file__))
 csv_file_path = os.path.join(current_dir, "../../../database/output.csv")
 
+from sklearn.model_selection import train_test_split, KFold
+from sklearn.exceptions import NotFittedError
+
+def safe_train_test_split(inputs, target, test_size=0.2, random_state=48):
```
Comment (akshatoza1) [inline backend/ml_model/use_cases/model.py:27]: Can you explain these changes, with the 2 functions that you added?

Comment (armaganngul) [reply backend/ml_model/use_cases/model.py:27]: Oh yeah my bad forgot to mention that. It's making sure the split is never bigger than the sample size so the model doesn't crash.

Comment (Jai0212) [reply backend/ml_model/use_cases/model.py:27]: That's a good addition as the model was crashing is some cases where the data we had in output.csv for less.

**Comment by Jai0212** [comment]
Everything seems fine with proper error handling. We need test coverage for this so that's something that needs to be looked into.

**Pull request merged**